src/components/Makefile.comp

```diff
@@ -35,6 +39,7 @@ SHARED_FLAGS=-fno-merge-constants -nostdinc -nostdlib -fno-pic
 OPT= -g
 #OPT= -O3
 CFLAGS=-m32 -D__x86__ -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -fno-stack-protector -fno-omit-frame-pointer -Wno-unused-variable -fvar-tracking $(INC_PATH) $(MUSLINC) $(LWIPINC) $(LUAINC) $(OPT) $(SHARED_FLAGS)
+CXXFLAGS=-fno-exceptions -fno-threadsafe-statics -Wno-write-strings -fpermissive $(CFLAGS)
```

Comment:
Why do we need `-fpermissive` in this Makefile? Isn't this preventing any component from getting the full range of errors?